test/Api.Test/SecretsManager/Controllers/ServiceAccountsControllerTests.cs

```diff
@@ -90,12 +95,70 @@ await sutProvider.GetDependency<ICreateServiceAccountCommand>().DidNotReceiveWit
                 Arg.Any<IEnumerable<IAuthorizationRequirement>>()).ReturnsForAnyArgs(AuthorizationResult.Success());
+
+        sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(Arg.Is(organizationId)).ReturnsNull();
+        sutProvider.GetDependency<IUserService>().GetProperUserId(default).ReturnsForAnyArgs(Guid.NewGuid());
+        var resultServiceAccount = data.ToServiceAccount(organizationId);
+        sutProvider.GetDependency<ICreateServiceAccountCommand>().CreateAsync(default, default)
+            .ReturnsForAnyArgs(resultServiceAccount);
+
+        await Assert.ThrowsAsync<NotFoundException>(() => sutProvider.Sut.CreateAsync(organizationId, data));
+        await sutProvider.GetDependency<ICreateServiceAccountCommand>().DidNotReceiveWithAnyArgs()
+            .CreateAsync(Arg.Any<ServiceAccount>(), Arg.Any<Guid>());
+    }
+
+    [Theory]
+    [BitAutoData(0)]
+    [BitAutoData(1)]
+    [BitAutoData(2)]
+    public async void CreateServiceAccount_NewServiceAccountSlotsRequired_CheckCallsUpdate(int newSlotsRequired, SutProvider<ServiceAccountsController> sutProvider,
+        ServiceAccountCreateRequestModel data, Guid organizationId, Organization mockOrg)
+    {
+        mockOrg.Id = organizationId;
+        sutProvider.GetDependency<IAuthorizationService>()
+            .AuthorizeAsync(Arg.Any<ClaimsPrincipal>(), data.ToServiceAccount(organizationId),
+                Arg.Any<IEnumerable<IAuthorizationRequirement>>()).ReturnsForAnyArgs(AuthorizationResult.Success());
+        sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(Arg.Is(organizationId)).Returns(mockOrg);
+        sutProvider.GetDependency<ICountNewServiceAccountSlotsRequiredQuery>()
+            .CountNewServiceAccountSlotsRequiredAsync(Arg.Is(organizationId), Arg.Is(1))
+            .ReturnsForAnyArgs(newSlotsRequired);
+        sutProvider.GetDependency<IUserService>().GetProperUserId(default).ReturnsForAnyArgs(Guid.NewGuid());
+        var resultServiceAccount = data.ToServiceAccount(organizationId);
+        sutProvider.GetDependency<ICreateServiceAccountCommand>().CreateAsync(default, default)
+            .ReturnsForAnyArgs(resultServiceAccount);
+
+        await sutProvider.Sut.CreateAsync(organizationId, data);
+        await sutProvider.GetDependency<ICreateServiceAccountCommand>().Received(1)
+            .CreateAsync(Arg.Any<ServiceAccount>(), Arg.Any<Guid>());
+
+        if (newSlotsRequired > 0)
+        {
+            await sutProvider.GetDependency<IUpdateSecretsManagerSubscriptionCommand>().Received(1)
+                .AdjustServiceAccountsAsync(Arg.Is(mockOrg), Arg.Is(newSlotsRequired));
+        }
+        else
+        {
+            await sutProvider.GetDependency<IUpdateSecretsManagerSubscriptionCommand>().DidNotReceiveWithAnyArgs()
+                .AdjustServiceAccountsAsync(Arg.Any<Organization>(), Arg.Any<int>());
+        }
```

Comment:
Suggestion only: I try to avoid conditional assertions like this, as well as test logic that effectively reimplements the sut. I think both make tests less reliable.

I assume you're trying to avoid repeating the setup code, but in that case I think a better solution is to extract it to a helper method. Then you can have separate tests: `WhenAutoscalingRequired_CallsUpdateSubscription` and `WhenAutoscalingNotRequired_DoesNotCallUpdateSubscription`, which both call the helper method in the arrange phase.